shanoir-ng-front/src/app/studies/study-list/study-list.component.ts

```diff
@@ -145,21 +153,60 @@ export class StudyListComponent extends BrowserPaginEntityListComponent<Study> {
         );
     }
 
+    private fetchStudyUsers(study: any): Promise<StudyUser[]> {
+        return this.studyService.getStudyUserFromStudyId(study.id).then(studyUsers => {
+            return studyUsers;
+        });
+    }
+
+    private fetchHasDUA(study: any): Promise<void> {
+        return this.studyService.hasDUAByStudyId(study.id).then(hasDUA => {
+            this.hasDUA = hasDUA;
+        });
+    }
+
     goToViewFromEntity(study: any): void {
-        if (study instanceof Study) {
+
+        Promise.all([
+            this.fetchHasDUA(study),
+            this.fetchStudyUsers(study),
+        ]).then(([boolean, studyUsers]) => {
+            for (let su of studyUsers) {
+                if (su.userId == this.keycloakService.getUserId()) {
+                    this.accessRequestValidated = true;
+                    if (su.confirmed) {
+                        this.isSuConfirmed = true;
```

Comment:
I think you can do that in one line without a if:
`this.isSuConfirmed = su.confirmed;`